spec/model/postgres/postgres_timeline_spec.rb

```diff
@@ -81,13 +81,13 @@
   describe "#latest_backup_label_before_target" do
     it "returns most recent backup before given target" do
       skip_if_frozen
-      stub_const("Backup", Struct.new(:last_modified))
+      backup = Struct.new(:key, :last_modified)
       most_recent_backup_time = Time.now
       expect(postgres_timeline).to receive(:backups).and_return(
         [
-          instance_double(Backup, key: "basebackups_005/0001_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 200),
-          instance_double(Backup, key: "basebackups_005/0002_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 100),
-          instance_double(Backup, key: "basebackups_005/0003_backup_stop_sentinel.json", last_modified: most_recent_backup_time)
+          instance_double(backup, key: "basebackups_005/0001_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 200),
+          instance_double(backup, key: "basebackups_005/0002_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 100),
+          instance_double(backup, key: "basebackups_005/0003_backup_stop_sentinel.json", last_modified: most_recent_backup_time)
```

Comment:
hm, if we're using a ginned up struct here, is verifying doubles (instance_double) really doing anything, or should we use regular "double"? Maybe I need to read more, but I'm a bit perplexed why the type value needs stubbing at all.

Reply:
I'm not sure, as I don't have much experience with rspec doubles.  I know the constant isn't needed.  I would guess that you could just return `backup` objects instead of `instance_double` or regular `double`, but I haven't tried.

Reply:
instance_double checks the base type has those methods, a bit like how this affects expect...returns:


```ruby
    # Prevents you from mocking or stubbing a method that does not exist on
    # a real object. This is generally recommended, and will default to
    # `true` in RSpec 4.
    mocks.verify_partial_doubles = true
```

But when the type is being ginned up, it mostly defeats the point. I guess it prevents a typo in regular `double`. Incidentally....the verbosity of the two names would be better swapped, but, path dependency.

Reply:
Okay, the reason for this is the backup is intended to the result object from an S3 query. The normal way to go about this is instance_double that kind of object (even though it would pull in the library at least a little). Unless it were too big and slow and we wanted to say "nah we're okay not checking the real implemention's methods"

Reply:
Here's a patch that normalifies these things:

```diff
diff --git a/spec/model/postgres/postgres_timeline_spec.rb b/spec/model/postgres/postgres_timeline_spec.rb
index bd310f9b..d0174276 100644
--- a/spec/model/postgres/postgres_timeline_spec.rb
+++ b/spec/model/postgres/postgres_timeline_spec.rb
@@ -80,13 +80,12 @@
 
   describe "#latest_backup_label_before_target" do
     it "returns most recent backup before given target" do
-      backup = Struct.new(:key, :last_modified)
       most_recent_backup_time = Time.now
       expect(postgres_timeline).to receive(:backups).and_return(
         [
-          instance_double(backup, key: "basebackups_005/0001_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 200),
-          instance_double(backup, key: "basebackups_005/0002_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 100),
-          instance_double(backup, key: "basebackups_005/0003_backup_stop_sentinel.json", last_modified: most_recent_backup_time)
+          instance_double(Minio::Client::Blob, key: "basebackups_005/0001_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 200),
+          instance_double(Minio::Client::Blob, key: "basebackups_005/0002_backup_stop_sentinel.json", last_modified: most_recent_backup_time - 100),
+          instance_double(Minio::Client::Blob, key: "basebackups_005/0003_backup_stop_sentinel.json", last_modified: most_recent_backup_time)
         ]
       )
 
@@ -122,11 +121,10 @@
   end
 
   it "returns list of backups" do
-    backup = Struct.new(:key)
     expect(postgres_timeline).to receive(:blob_storage).and_return(instance_double(MinioCluster, url: "https://blob-endpoint", root_certs: "certs")).at_least(:once)
 
     minio_client = Minio::Client.new(endpoint: "https://blob-endpoint", access_key: "access_key", secret_key: "secret_key", ssl_ca_file_data: "data")
-    expect(minio_client).to receive(:list_objects).with(postgres_timeline.ubid, "basebackups_005/").and_return([instance_double(backup, key: "backup_stop_sentinel.json"), instance_double(backup, key: "unrelated_file.txt")])
+    expect(minio_client).to receive(:list_objects).with(postgres_timeline.ubid, "basebackups_005/").and_return([instance_double(Minio::Client::Blob, key: "backup_stop_sentinel.json"), instance_double(Minio::Client::Blob, key: "unrelated_file.txt")])
     expect(Minio::Client).to receive(:new).and_return(minio_client)
 
     expect(postgres_timeline.backups.map(&:key)).to eq(["backup_stop_sentinel.json"])
```